anymail/backends/sendinblue.py

```diff
@@ -0,0 +1,144 @@
+
+    def get_api_endpoint(self):
+        return "smtp/email"
+
+    def init_payload(self):
+        self.data = {  # becomes json
+            'to': [],
+            'cc': [],
+            'bcc': [],
+            'sender': [],
+        }
+
+    def serialize_data(self):
+        """Performs any necessary serialization on self.data, and returns the result."""
+
+        headers = self.data["headers"]
+        if "Reply-To" in headers:
+            # Reply-To must be in its own param
+            reply_to = headers.pop('Reply-To')
+            self.set_reply_to(parse_address_list([reply_to]))
+        if len(headers) > 0:
+            self.data["headers"] = dict(headers)  # flatten to normal dict for json serialization
+        else:
+            del self.data["headers"]  # don't send empty headers
+
+        return self.serialize_json(self.data)
+
+    #
+    # Payload construction
+    #
+
+    @staticmethod
+    def email_object(email):
+        """Converts EmailAddress to SendinBlue API array"""
+        email_object = dict()
+        email_object['email'] = email.addr_spec
+        if email.display_name:
+            email_object['name'] = email.display_name
+        return email_object
+
+    def set_from_email(self, email):
+        self.data["sender"] = self.email_object(email)
+
+    def set_recipients(self, recipient_type, emails):
+        assert recipient_type in ["to", "cc", "bcc"]
+        if emails:
+            # Normally, exactly one "personalizations" entry for all recipients
+            # (Exception: with merge_data; will be burst apart later.)
+            self.data[recipient_type] = [self.email_object(email) for email in emails]
+            self.all_recipients += emails  # used for backend.parse_recipient_status
+
+    def set_subject(self, subject):
+        if subject != "":  # see note in set_text_body about template rendering
+            self.data["subject"] = subject
+
+    def set_reply_to(self, emails):
+        # SendinBlue only supports a single address in the reply_to API param.
+        if len(emails) > 1:
+            self.unsupported_feature("multiple reply_to addresses")
+        if len(emails) > 0:
+            self.data['replyTo'] = self.email_object(emails[0])
+
+    def set_extra_headers(self, headers):
+        self.data['headers'] = [header for header in headers]
+
+    def set_text_body(self, body):
+        self.data['textContent'] = body
+
+    def set_html_body(self, body):
+        if body is not None:
+            if "htmlContent" in self.data:
+                self.unsupported_feature("multiple html parts")
+
+            self.data['htmlContent'] = body
+
+    def add_attachment(self, attachment):
+        """Converts attachments to SendinBlue API {name, base64} array"""
+
+        att = {
+            'name': attachment.name,
+            'content': attachment.b64content,
+        }
+
+        self.data.setdefault("attachment", []).append(att)
```

Comment:
Every payload should implement `set_esp_extra` to allow overrides and esp-specific parameters. Usually just dict.update into self.data works fine. (Anymail utils.update_deep can also be used.) [Example](https://github.com/anymail/django-anymail/blob/v1.4/anymail/backends/mailgun.py#L198-L199).

Reply:
Done. I've found no one information for the extra header on the v3 so i've not included some check in the code.

> Note:
“X-Mailin-IP” will not be sent along with the mail headers in original email.
“X-param1″, “X-param2″, “X-Mailin-custom” and “X-Mailin-Tag” will be sent along with the mail headers in original email.
“X-Mailin-custom” and “X-Mailin-Tag” will also be sent along with the JSON response of webhooks, when some event occurs. Also, the “X-Mailin-Tag” will be received as only “tag” in the JSON response.

[API v2 reference](https://apidocs.sendinblue.com/tutorial-sending-transactional-email/)